src/resolvers/user.ts

```diff
@@ -1,36 +1,322 @@
+      }
+      // Unauthenticated
+      throw AuthenticationError();
+    },
+
+    // Change the current user's password
+    updatePassword: async (_, { oldPassword, newPassword }, context: MyContext): Promise<User> => {
+      if (isAuthorized(context?.token)) {
+        const user = await User.findById('updatePassword resolver', context, context.token.id);
+        // Only continue if the user is active and not locked
+        if (!user || !user.active || user.locked) {
+          throw ForbiddenError();
+        }
+
+        const updated = await new User(user).updatePassword(context, oldPassword, newPassword);
+        if (updated) {
+          return updated;
+        }
+      }
+      // Unauthenticated
+      throw AuthenticationError();
+    },
+
+    // Deactivate the specified user Account (Admin only)
+    deactivateUser: async (_, { userId }, context: MyContext): Promise<User> => {
+      const ref = 'deactivateUser resolver';
+      if (isAdmin(context.token)) {
+        const result = await User.findById(ref, context, userId);
+        // For some reason these are being returned a Objects and not User!
+        const user = new User(result);
+        // Only continue if the current user's affiliation matches the user OR they are SuperAdmin
+        if (context.token.affiliationId === user.affiliationId || isSuperAdmin(context.token)) {
+          user.active = false;
+          const updated = await User.update(context, new User(user).tableName, user, ref, ['password']);
+
+          if (!updated) {
+            throw InternalServerError('Unable to deactivate the user at this time');
+          }
+          // Return the result, because updated will not return a User since they are now inactive
+          return result as User;
+        }
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
+    },
+    // Reactivate the specified user Account (Admin only)
+    activateUser: async (_, { userId }, context: MyContext): Promise<User> => {
+      const ref = 'activateUser resolver';
+      if (isAdmin(context.token)) {
+        const result = await User.findById(ref, context, userId);
+        // For some reason these are being returned a Objects and not User!
+        const user = new User(result);
+        // Only continue if the current user's affiliation matches the user OR they are SuperAdmin
+        if (context.token.affiliationId === user.affiliationId || isSuperAdmin(context.token)) {
+          user.active = true;
+          const updated = await User.update(context, user.tableName, user, ref, ['password']);
+
+          if (!updated) {
+            throw InternalServerError('Unable to activate the user at this time');
+          }
+          return result as User;
+        }
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
+    },
+    // Merge the 2 user accounts (Admin only)
```

Comment:
L289: Should the comment also include "SuperAdmin", or is it understood that "Admin only" captures both Admin and SuperAdmin?